src/13-ref2vec/03-user.ts

```diff
@@ -0,0 +1,137 @@
+import { ARTWORK_IDS } from "./ids/artwork-ids"
+import { ARTIST_IDS } from "./ids/artist-ids"
+import weaviate, { generateUuid5 } from "weaviate-ts-client"
+import _ from "lodash"
+import dotenv from "dotenv"
+import { ClassName, Objects, ReferenceProperty, User } from "./types/types"
+
+// Config
+dotenv.config()
+
+// Constants
+const CLASS_NAME: ClassName = "Users"
+const BATCH_SIZE: number = 10
+const SAMPLE_SIZE: number = 3
+const USERS: User[] = [{ id: "abc123", name: "Percy Cat" }]
+
+const client = weaviate.client({
+  scheme: "https",
+  host: "https://weaviate.stg.artsy.systems",
+})
+
+async function main() {
+  const sampleArtworkIds = _.shuffle(ARTWORK_IDS).slice(0, SAMPLE_SIZE)
+  const sampleArtistIds = _.shuffle(ARTIST_IDS).slice(0, SAMPLE_SIZE)
+
+  await prepareCollection(CLASS_NAME)
+  await insertObjects(USERS, BATCH_SIZE)
+  await createReferences(
+    USERS,
+    sampleArtworkIds,
+    "SmallNewTrendingArtworks",
+    "likedArtworks"
+  )
+  await createReferences(
+    USERS,
+    sampleArtistIds,
+    "SmallNewTrendingArtists",
+    "likedArtists"
+  )
+}
+
+main()
+
+async function prepareCollection(className: ClassName) {
+  const client = weaviate.client({
+    scheme: "https",
+    host: "https://weaviate.stg.artsy.systems",
+  })
+
+  const alreadyExists = await client.schema.exists(className)
+
+  if (alreadyExists) {
+    console.log(`${className} class already exists, deleting it`)
+    await client.schema.classDeleter().withClassName(className).do()
+  }
+
+  const classSchema = {
+    class: className,
+    moduleConfig: {
+      "ref2vec-centroid": {
+        referenceProperties: ["likedArtworks", "likedArtists"],
+      },
+    },
+    vectorizer: "ref2vec-centroid",
```

Comment:
Since the `User` object's vector is calculated from the vectors of the objects it has references to, it won't have a vector until we create  references. So L28-L40 in this file.